### PR TITLE
Plugins: Do not fail bootstrap stage if single decorate step fails

### PR DESCRIPTION
**Pull request by wbrowne**
**What is this feature?**

Ensures that a failed decorate step for one plugin, does not fail the entire stage

**Why do we need this feature?**

Consistency across loader pipeline stages

**Who is this feature for?**

Plugins Platform

**Special notes for your reviewer:**
This also tries to make log parameters naming consistent:
- `"pluginID"` -> `"pluginId"` (prefer camel case)
- `"err"` -> `"error"` (prefer using full words)

Please check that:
- [X] It works as expected from a user's perspective.
- [X] If this is a pre-GA feature, it is behind a feature toggle.
- [X] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: pkg/plugins/manager/pipeline/initialization/initialization.go [hunk 1]
```diff
@@ -54,7 +54,7 @@ func (i *Initialize) Initialize(ctx context.Context, ps []*plugins.Plugin) ([]*p
 			ip, err = init(ctx, p)
 			if err != nil {
 				stepFailed = true
-				i.log.Error("Could not initialize plugin", "pluginId", p.ID, "err", err)
+				i.log.Error("Could not initialize plugin", "pluginID", p.ID, "err", err)
```
Comment (marefr) [inline pkg/plugins/manager/pipeline/initialization/initialization.go:57]: Nit. Prefer using camelCase when logging https://github.com/grafana/grafana/blob/main/contribute/backend/instrumentation.md#naming-conventions

```suggestion
				i.log.Error("Could not initialize plugin", "pluginId", p.ID, "error", err)
```

File: pkg/plugins/manager/pipeline/initialization/steps.go [hunk 2]
```diff
@@ -80,7 +80,7 @@ func newBackendProcessStarter(processManager process.Service) *BackendClientStar
 // Start will start the backend plugin process.
 func (b *BackendClientStarter) Start(ctx context.Context, p *plugins.Plugin) (*plugins.Plugin, error) {
 	if err := b.processManager.Start(ctx, p.ID); err != nil {
-		b.log.Error("Could not start plugin", "pluginId", p.ID, "err", err)
+		b.log.Error("Could not start plugin", "pluginID", p.ID, "err", err)
```
Comment (marefr) [inline pkg/plugins/manager/pipeline/initialization/steps.go:83]:
```suggestion
		b.log.Error("Could not start plugin", "pluginId", p.ID, "error", err)
```

File: pkg/plugins/manager/pipeline/termination/steps.go [hunk 3]
```diff
@@ -64,7 +64,7 @@ func newBackendProcessTerminator(processManager process.Service) *BackendProcess
 
 // Terminate stops a backend plugin process.
 func (t *BackendProcessTerminator) Terminate(ctx context.Context, p *plugins.Plugin) error {
-	t.log.Debug("Stopping plugin process", "pluginId", p.ID)
+	t.log.Debug("Stopping plugin process", "pluginID", p.ID)
```
Comment (marefr) [inline pkg/plugins/manager/pipeline/termination/steps.go:67]:
```suggestion
	t.log.Debug("Stopping plugin process", "pluginId", p.ID)
```

File: pkg/plugins/manager/pipeline/termination/steps.go [hunk 4]
```diff
@@ -92,7 +92,7 @@ func (d *Deregister) Deregister(ctx context.Context, p *plugins.Plugin) error {
 	if err := d.pluginRegistry.Remove(ctx, p.ID); err != nil {
 		return err
 	}
-	d.log.Debug("Plugin unregistered", "pluginId", p.ID)
+	d.log.Debug("Plugin unregistered", "pluginID", p.ID)
```
Comment (marefr) [inline pkg/plugins/manager/pipeline/termination/steps.go:95]:
```suggestion
	d.log.Debug("Plugin unregistered", "pluginId", p.ID)
```

File: pkg/plugins/manager/pipeline/validation/validation.go [hunk 5]
```diff
@@ -41,27 +40,26 @@ func New(cfg *config.Cfg, opts Opts) *Validate {
...
 		stepFailed := false
-		for _, validate := range t.validateSteps {
-			err = validate(ctx, p)
-			if err != nil && !errors.Is(err, nil) {
+		for _, validate := range v.validateSteps {
+			err := validate(ctx, p)
+			if err != nil {
 				stepFailed = true
-				t.log.Error("Plugin verification failed", "pluginID", p.ID, "err", err)
+				v.log.Error("Plugin validation failed", "pluginID", p.ID, "err", err)
```
Comment (marefr) [inline pkg/plugins/manager/pipeline/validation/validation.go:55]:
```suggestion
				v.log.Error("Plugin validation failed", "pluginId", p.ID, "error", err)
```

**Review by marefr — APPROVED** [review]
LGTM with some suggestions

File: pkg/plugins/manager/pipeline/initialization/initialization.go (hunk 1 above)
Comment (wbrowne) [reply pkg/plugins/manager/pipeline/initialization/initialization.go]: Had no idea! Have gone ahead and tried fix across the entire `pkg/plugins`!

**Review by xnyo — APPROVED** [review]
Great stuff! :clap: :clap: 

I found a couple of instances of `pluginID` left, it would be great to change those as well, but LGTM otherwise!

https://github.com/grafana/grafana/blob/72da44db0e62dfcf7d9ea93253fba2fd0eacf5bc/pkg/services/pluginsintegration/pipeline/steps.go#L44

https://github.com/grafana/grafana/blob/72da44db0e62dfcf7d9ea93253fba2fd0eacf5bc/pkg/services/pluginsintegration/pipeline/steps.go#L73

https://github.com/grafana/grafana/blob/72da44db0e62dfcf7d9ea93253fba2fd0eacf5bc/pkg/services/pluginsintegration/pipeline/steps.go#L112

**Review by marefr — APPROVED** [review]
LGTM

**Pull request merged**